### ISSUE TITLE
patches numberformat does not work for arbitrary representations ( broken with iso4217 CHF)

### ISSUE THREAD

**Issue by simonfloris** [awesto/django-shop#408]
patches.numberformat.py line 37
Example:

str_number   : u'SFr. 17.10'  

``````
    if '.' in str_number:
        int_part, dec_part = str_number.split('.')

```=> too many values to unpack

``````

**Comment by jrief** [comment]
The input for ``str_number`` should not contain the ``SFr.`` part. Which function is calling this format function? The error must be there.

**Comment by jrief** [comment]
Since that monkey patch is required only for Django-1.8, how about upgrading your project to Django-1.9/1.10?

**Comment by simonfloris** [comment]
Sadly, as we are already updating from django 1.5 to the latest LTS, which is 1.8,  an even further update is not doable.

The shop is a rather small part of the application and is already eating up a lot of the migration-time as it is.

The shop is now integrated without the rest-api and the additional technical debt(angular,bootstrap,cascde) that comes with the shop update from 0.2.
We "just" migrated to the new models and the workflow. The Currency/Money-approach I'd like to keep, too.

The error occurs, when rendering the price e.g. cart_item.line_total in a standard django template

Alternatively, is there a way to override the iso4217 settings?

**Comment by jrief** [comment]
It is strange that the function ``number_format`` is invoked with a value of type string. This should be a Decimal. Please also check in django.utils.numberformat.py: ``localize()`` which checks the type.

Since MoneyInCHF inherits from Decimal, I can use that utility from Django.

BTW: please paste a stack-trace using dpaste.de or similar services.

**Comment by simonfloris** [comment]
Sorry, I could not investigate this further, I removed the Symbol from the MONEY_FORMAT settings and reused some already existing taghelpers.

But it is still unclear how to override or at least extend the  iso4217.CURRENCIES from outside the app?

**Comment by jrief** [comment]
this has been fixed in PR #551 and will be part of version 0.10.2
